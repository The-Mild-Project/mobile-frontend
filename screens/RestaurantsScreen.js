import React, { useState } from 'react';
import { FlatList, Image, Platform, StyleSheet, Text, TouchableOpacity, View, Button, AsyncStorage } from 'react-native';
import {ScrollView} from "react-native-gesture-handler";
import useRestaurants from "../hooks/useRestaurants";
import ResultsList from "../components/ResultsList";

// TODO: Add Searchbar, add term to search
const RestaurantsScreen = () => {
    const [term, setTerm] = useState('');
    // returns an array
    const [searchApi, results] = useRestaurants();


    // so we can filter by type of food later
    const filterResultsByType = (genre) => {
        return results.filter(result => {
            return result.genre === genre;
        })
    };


    return (
        <ScrollView style={styles.container} contentContainerStyle={styles.contentContainer}>
            <View style={styles.container}>
                <Text>Restaurants</Text>
                <Text>Search Will Go Here</Text>
            </View>
            <ResultsList title="First" results={results}/>
        </ScrollView>
    )
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        paddingTop: 22
    },
    item: {
        padding: 10,
        fontSize: 18,
        height: 44,
    },
});

export default RestaurantsScreen;