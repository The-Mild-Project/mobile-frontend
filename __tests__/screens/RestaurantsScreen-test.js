import * as React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import RestaurantsScreen from '../../screens/RestaurantsScreen';

jest.mock('expo', () => ({
    AppLoading: 'AppLoading',
}));

describe('App', () => {
    jest.useFakeTimers();

    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it(`renders the loading screen`, () => {
        const tree = renderer.create(<RestaurantsScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
