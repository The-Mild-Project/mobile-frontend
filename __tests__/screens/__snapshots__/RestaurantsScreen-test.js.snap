// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the loading screen 1`] = `
<RCTScrollView
  collapsable={false}
  contentContainerStyle={
    Object {
      "paddingTop": 30,
    }
  }
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Object {
      "flex": 1,
      "paddingTop": 12,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fefefe",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 50,
          "marginHorizontal": 30,
          "marginTop": 10,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Search Food"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#fefefe",
            "borderWidth": 1,
            "flex": 1,
            "fontSize": 18,
            "marginHorizontal": 15,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View>
      <Text />
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;
